editor/filesystem_dock.cpp

```diff
@@ -2120,6 +2147,63 @@ void FileSystemDock::set_file_list_display_mode(FileListDisplayMode p_mode) {
 	_toggle_file_display();
 }
 
+Control *FileSystemDock::create_tooltip_for_path(const String &p_path) const {
+	if (DirAccess::exists(p_path)) {
+		// No tooltip for directory.
+		return nullptr;
```

Comment:
Well, this is kind of hacky and you know how this story continues in Godot. After the PR is merged, you get a proposal opened:

"Nice resource file info tooltip you have there, can I do this for my custom resource type from my extension?" 

So I think this should be a bit more pluggable. So maybe it would be good to make a tiny class: `EditorResourceFileTooltipPlugin` and register them directly into the FileSystem dock? (via EditorPlugin API).

Reply:
That's too much work, I'd rather wait for someone to actually request it (the code won't be difficult to replace).